src/main/java/core/basesyntax/strategy/BalanceOperation.java

```
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> stock) {
+        stock.put(transaction.getFruit(), transaction.getQuantity());
```

This method does not validate null parameters. The checklist requires: "Null parameters (e.g., null file path, null list inputs) should be validated and cause informative exceptions." Add checks for `transaction` and `stock` (and for `transaction.getFruit()` if needed) and throw an informative RuntimeException if any are null instead of letting a NullPointerException occur.